graphdatascience/procedure_surface/cypher/wcc_proc_runner.py

```diff
@@ -0,0 +1,261 @@
+from typing import Any, List, Optional
+
+from pandas import DataFrame
+
+from ...call_parameters import CallParameters
+from ...graph.graph_object import Graph
+from ...query_runner.query_runner import QueryRunner
+from ..api.wcc_endpoints import WccEndpoints, WccMutateResult, WccStatsResult, WccWriteResult
+
+
+class WccCypherEndpoints(WccEndpoints):
+    """
+    Implementation of the WCC algorithm endpoints.
+    This class handles the actual execution by forwarding calls to the query runner.
+    """
+
+    def __init__(self, query_runner: QueryRunner):
+        self._query_runner = query_runner
+
+    def mutate(
+        self,
+        G: Graph,
+        mutate_property: str,
+        threshold: Optional[float] = None,
+        relationship_types: Optional[List[str]] = None,
+        node_labels: Optional[List[str]] = None,
+        sudo: Optional[bool] = None,
+        log_progress: Optional[bool] = None,
+        username: Optional[str] = None,
+        concurrency: Optional[int] = None,
+        job_id: Optional[str] = None,
+        seed_property: Optional[str] = None,
+        consecutive_ids: Optional[bool] = None,
+        relationship_weight_property: Optional[str] = None,
+    ) -> WccMutateResult:
+        # Build configuration dictionary from parameters
+        config: dict[str, Any] = {
+            "mutateProperty": mutate_property,
+        }
+
+        self._create_procedure_config(
+            config,
+            concurrency,
+            consecutive_ids,
+            job_id,
+            log_progress,
+            None,
+            node_labels,
+            relationship_types,
+            relationship_weight_property,
+            seed_property,
+            sudo,
+            threshold,
+            username,
+        )
+
+        # Run procedure and return results
+        params = CallParameters(graph_name=G.name(), config=config)
+        params.ensure_job_id_in_config()
+
+        cypher_result = self._query_runner.call_procedure(endpoint="gds.wcc.mutate", params=params).squeeze()
+
+        return WccMutateResult(
+            cypher_result["componentCount"],
+            cypher_result["componentDistribution"],
+            cypher_result["preProcessingMillis"],
+            cypher_result["computeMillis"],
+            cypher_result["postProcessingMillis"],
+            cypher_result["mutateMillis"],
+            cypher_result["nodePropertiesWritten"],
+            cypher_result["configuration"],
+        )
```

Comment:
i could also see how we just take the cypher_result and rename the dict keys to snake_case -> `WccMutateResult(**cypher_result)`

Reply:
That is a good idea, I added that.